src/shillelagh/backends/apsw/vt.py

```diff
@@ -429,10 +462,14 @@ def BestIndexObject(self, index_info: apsw.IndexInfo) -> bool:
         (
             constraints_used,
             index_number,
-            index_name,
+            indexes,
+            orderbys_to_process,
             orderby_consumed,
             estimated_cost,
-        ) = self.BestIndex(constraints, orderbys)
+        ) = self._build_index(constraints, orderbys)
+
+        requested_columns = sorted({column_names[i] for i in index_info.colUsed})
+        index_name = json.dumps([indexes, orderbys_to_process, requested_columns])
```

Comment:
Could use object top level rather than array for some safety.

Reply:
Good idea, that will also clean the type hints.

Reply:
You could also use a TypedDict here for some extra guards:
```python
index_data = BestIndexObjectIndexData(
  indexes=indexes, orderbys_to_process=orderbys_to_process, requested_columns=requested_columns
)
....
index_data: BestIndexObjectIndexData = json.loads(....)
```
... though that might not work given there are actually two different formats that you are serializing and instance of checks don't work on `TypedDict`